aggregateroot.go

```diff
@@ -23,6 +23,9 @@ const (
 // ErrAggregateAlreadyExists returned if the aggregateID is set more than one time
 var ErrAggregateAlreadyExists = errors.New("its not possible to set ID on already existing aggregate")
 
+// ErrAggregateNeedsToBeAPointer return if aggregate is sent in as value object
+var ErrAggregateNeedsToBeAPointer = errors.New("aggregate needs to be a pointer")
```

Comment:
Could this be generalized to an error saying it has to be a pointer type? For instance:
```suggestion
// ErrNonPointerType ...
var ErrNonPointerType = errors.New("type needs to be a pointer")
```

Reply:
I'm not sure if this would be valuable. Just raising the question!

Reply:
It's a good question but this error is only valid if an aggregate is sent in and if it's not a pointer. This check is not conducted on any other type.